include/boost/math/distributions/dirichlet.hpp

```diff
@@ -0,0 +1,405 @@
+inline bool check_variance(const char *function,
+                           const VectorType &variance,
+                           RealType *result,
+                           const Policy &pol)
+{
+  for (size_t i = 0; i < variance.size(); ++i)
+  {
+    if (!(boost::math::isfinite)(i) || (i <= 0))
+    {
+      *result = policies::raise_domain_error<RealType>(
+          function,
+          "variance argument is %1%, but must be > 0 !", i, pol);
+      return false;
+    }
+  }
+  return true;
+} // bool check_variance
+} // namespace dirichlet_detail
+
+template <class VectorType = std::vector<double>, class RealType = double, class Policy = policies::policy<>>
+class dirichlet_distribution
+{
+public:
+  dirichlet_distribution(VectorType concentration) : concentration(concentration)
+  {
+    RealType result;
+    dirichlet_detail::check_dist(
+        "boost::math::dirichlet_distribution<%1%>::dirichlet_distribution",
+        concentration,
+        &result, Policy());
+    sum_concentration = accumulate(concentration.begin(), concentration.end(), 0);
+  } // dirichlet_distribution constructor.
+
+  // Accessor functions:
+  VectorType Concentration() const
+  {
+    return concentration;
+  }
+
+  size_t Order() const
+  {
+    return concentration.size();
+  }
+
+  static VectorType find_concentration(
+      VectorType mean,     // Expected value of mean.
+      VectorType variance) // Expected value of variance.
+  {
+    assert(("Dimensions of mean and variance must be same!", mean.size() == variance.size()));
+    static const char *function = "boost::math::dirichlet_distribution<%1%>::find_concentration";
+    RealType result = 0; // of error checks.
+    if (!(dirichlet_detail::check_mean(function, mean, &result, Policy()) && dirichlet_detail::check_variance(function, variance, &result, Policy())))
+    {
+      return result;
+    }
+    VectorType c;
+    for (size_t i = 0; i < mean.size(); ++i)
+    {
+      c.push_back(mean[i] * (((mean[i] * (1 - mean[i])) / variance[i]) - 1));
+    }
+    return c;
+  } // RealType find_concentration
+
+  // TODO
+  // static VectorType find_concentration(
+  //     VectorType x,           //  x.
+  //     VectorType probability) // cdf
+  // {
+  //   assert(("", x.size() == probability.size()));
+  //   static const char *function = "boost::math::dirichlet_distribution<%1%>::find_conentration";
+  //   RealType result = 0; // of error checks.
+  //   if (!(dirichlet_detail::check_prob(function, probability, &result, Policy()) && dirichlet_detail::check_x(function, x, &result, Policy())))
+  //   {
+  //     return result;
+  //   }
+  //   return ;
+  // } // RealType find_concentration(x, probability)
+
+private:
+  VectorType concentration; // https://en.wikipedia.org/wiki/Concentration_parameter.
+  RealType sum_concentration;
+}; // template <class RealType, class Policy> class dirichlet_distribution
+
+
+template <class VectorType, class RealType, class Policy>
```

Comment:
Again, `template<class RandomAccessContainer>` and then `using Real = typename RandomAccessContainer::value_type`.